doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1023 @@
+
+![Remote MAC Handling](images/remotemac.PNG "Figure : Remote MAC handling")
+__Figure 10: Remote MAC handling__
+
+
+#### 4.3.7.3 MAC Move handling
+
+##### Local to Local
+
+There is no special handling by SwSS. 
+
+Local to Local MAC moves will be identified by BGP and no MAC route will be re-originated. 
+
+##### Remote to Local
+
+For MAC move cases from remote to local, fdbsyncd removes the 
+remote entry from VXLAN_FDB_TABLE. The fdborch should not delete the 
+MAC from the internal cache and the hardware. 
+
+##### Local to Remote
+
+MAC move from local to remote tunnel is handled as follows.
+
+- fdbsyncd adds the new remote MAC entry to the APP_VXLAN_FDB_TABLE.
+- fdborch looks up internal cache to see if an entry exists. If it exists with a local port as the destination then the entry is deleted and the new entry populated.
+- If the local MAC is configured as static then the received remote MAC will not be programmed and instead will be marked as pending. This will be programmed when the local static MAC is removed. 
+
+##### Remote to Remote
+
+MAC Move from remote tunnel-1 to remote tunnel-2 is handled as follows. 
+
+- On identifying a MAC Move, BGP deletes the old entry from APP_VXLAN_FDB_TABLE and adds the new entry pointing to the tunnel-2.
+- fdborch processing is same as for a regular remote MAC add/delete.  
+
+
+
+
+### 4.3.8 MAC-IP route handling
+
+Local MAC-IP bindings (ARP/ND entries) will be directly learnt by BGP (Zebra) from the IP stack.
+
+![MAC-IP Handling](images/type2route.PNG "Figure : MAC-IP handling")
+__Figure 11: EVPN MAC-IP Route handling__
+
+The remote MAC-IP routes will be installed by BGP (Zebra) in Linux neighbor table against Vlan netdevice. Neighsyncd subscribes to neighbor entries and will receive remote MAC-IP bindings as well. These entries will continue to go into NEIGH_TABLE in APP_DB, and neighorch will update SAI database.
+
+
+
+### 4.3.9 IP Prefix (Type-5) route handling
+
+BGP VRF routes will be exported into EVPN as IP prefix routes if corresponding VRF is configured with L3VNI and corresponding IRB VLAN is present and belongs to the corresponding VRF.  The remote EVPN IP prefix routes are imported into BGP VRF table and after path selection downloaded to Linux and Fpmsyncd.
+
+![EVPN IP Prefix route handling](images/type5route.PNG "Figure 11: EVPN_Type5_Route")
+__Figure 12: EVPN_Type5_Route__
+
+#### VRF Route Table with Tunnel Nexthop
+
+The connected prefixes bound to VNET go into VNET_ROUTE_TABLE, and remote prefixes are programmed in VNET_ROUTE_TUNNEL_TABLE.
+
+In contrast to the above, VRF routes are programmed in ROUTE_TABLE:{{vrf_name}} table including local (connected, static) and remote (BGP, IGP) VRF-lite prefixes.
+
+In order to program VRF routes imported from EVPN Type-5 prefix routes, <endpoint, router-MAC, and VNI> values are required along with the prefix in the route entry.
```

Comment:
Please describe a little more about how router-mac is derived.

Reply:
Interface MAC of IRB router interface is used as router-mac.
L3 VNI will be associated with a Vlan interface (IRB router interface), Vlan interface MAC of that associated IRB interface is used as router-mac.

Reply:
What happens in-case of MCLAG use case? How is the router-mac derived?